jetty-core/jetty-osgi/src/main/java/org/eclipse/jetty/osgi/util/Util.java

```diff
@@ -61,22 +60,24 @@ public static URI resolvePathAsLocalizedURI(String path, Bundle bundle, Path jet
         if (StringUtil.isBlank(path))
             return null;
 
-        if (path.startsWith("/") || path.startsWith("file:/")) //absolute location
-            return URIUtil.toURI(path);
-        else
+        if (path.startsWith("file:/"))
+            return new URI(path);
+
+        if (path.startsWith("/") && File.pathSeparatorChar != '/')
```

Comment:
Not sure I understand the second condition.
What is that hoping to address?

Reply:
It is broken!... it should be just separatorChar.  
The idea is that if the file system uses '/' as the separator, then the Path can deal with it below.  If we see a '/' and the file system is not using '/', then it needs to be handled as a URI.

Reply:
@joakime is that better?

Reply:
This would trip up things on Windows.
Take the input string `E:/code/base` as an example.
The `File.pathSeparatorChar` is `\`.
But that String also conforms to URI space.

```
jshell> var u1 = URI.create("E:/code/base")
u1 ==> E:/code/base

jshell> u1.getScheme()
$2 ==> "E"

jshell> u1.getPath()
$3 ==> "/code/base"

jshell> u1.getSchemeSpecificPart()
$4 ==> "/code/base"
```

And the Windows UNC syntax is equally tricky.

```
jshell> var u1 = URI.create("//machine/share/dir")
u1 ==> //machine/share/dir

jshell> u1.getScheme()
$2 ==> null

jshell> u1.isAbsolute()
$3 ==> false

jshell> u1.getPath()
$4 ==> "/share/dir"

jshell> u1.getAuthority()
$5 ==> "machine"
```

Reply:
Those are valid Strings for path references in Windows.
Can be used by `new File(String)` and even `Paths.of(String)` equally successfully.

Reply:
@joakime  I don't think think `E:/code/base` is a problem. It doesn't start with '/' so we will try it with `FileSystems.getDefault().getPath(path);`   So I think that is correct.

The line that you highlighted is only if somebody passed a path like `/foo/bar` in.   It starts with '/', but if we are on a file system that also uses '/', then it will be passed to `FileSystems.getDefault().getPath(path);`.  If it is on windows, then the separation char is not '/' so we will try `new URI("file:" + path)`.    So this is only for relative URI paths passed into this method on a windows system.

in short, this line is exactly so this method will work on windows.

(P.S. Not sure why this is not using Resource... but I don't know enough about OSGi resources to change this).